language/stdlib/modules/libra_account.move

```diff
@@ -297,16 +311,20 @@ module LibraAccount {
         account.delegated_key_rotation_capability = false;
     }
 
-    // Creates a new account at `fresh_address` with an initial balance of zero
-    // Creating an account at 0x0 will cause runtime failure as it is a
+    // Creates a new account at `fresh_address` with an initial balance of zero and authentication
+    // key `auth_key_prefix` | `fresh_address`
```

Comment:
Just out of curiosity why did we choose `auth_key_prefix | fresh_address` vs `hash(fresh_address)` as we did before?

Reply:
Basically: because `hash(fresh_address) != tx.pubkey` under the new scheme where address lengths and authe key lengths are different. The prologue checks `hash(tx.pubkey)` = `auth_key`. So the idea is that we choose `auth_key_prefix` as the first 16 bytes of `hash(tx.pubkey)` and `fresh_address` as the second 16 bytes.